cowprotocol/cowswap
# Add Create LlamaPay Vesting iframe hook

## Summary
Add Create LlamaPay Vesting hook
![image](https://github.com/user-attachments/assets/f6e41332-cf41-4714-a469-4e47acf11859)
![image](https://github.com/user-attachments/assets/160da105-83ca-4112-9baa-940ba796cc68)

## To Test
1. (Access `/#/100/swap/hooks/` running locally) or (access `https://dev.swap.cow.fi/#/100/swap/hooks/` and add custom post-hook with URL `https://cow-hooks-dapps-create-vesting.vercel.app`)
2. Connect your wallet and switch to a chain with deployed LlamaPay vesting factory (Mainnet, Gnosis or Arbitrum)
3. Fill some swap order
4. In Hooks > Add Post-Hook Action, the Create Vesting Hook should appear in the "All hooks" section
5. Fill the form data and add the post-hook

## Issues that are still being fixed:
- Tenderly simulations are currently not working
- If you try to manually input a vesting amount that is larger than your balance, the encoding will fail

## Other info
Old PR: https://github.com/cowprotocol/cowswap/pull/5026
